source/extensions/clusters/common/logical_host.h

```diff
@@ -9,66 +9,77 @@
 namespace Envoy {
 namespace Upstream {
 
+/**
+ * A logical family of hosts, supporting dynamic update. This shares much
+ * of its implementation with HostDescriptionImpl, but has non-const address
+ * member variables that are lock-protected.
+ */
+class LogicalHostDescription : public HostDescriptionImplBase {
+public:
+  LogicalHostDescription(
+      ClusterInfoConstSharedPtr cluster, const std::string& hostname,
+      Network::Address::InstanceConstSharedPtr dest_address, MetadataConstSharedPtr metadata,
+      const envoy::config::core::v3::Locality& locality,
+      const envoy::config::endpoint::v3::Endpoint::HealthCheckConfig& health_check_config,
+      uint32_t priority, TimeSource& time_source)
+      : HostDescriptionImplBase(cluster, hostname, dest_address, metadata, locality,
+                                health_check_config, priority, time_source),
+        address_(dest_address) {}
+
+  // HostDescription
+  Network::Address::InstanceConstSharedPtr healthCheckAddress() const override;
+
+protected:
+  // The first entry in the address_list_ should match the value in address_.
+  Network::Address::InstanceConstSharedPtr address_ ABSL_GUARDED_BY(address_lock_);
+  SharedConstAddressVector address_list_ ABSL_GUARDED_BY(address_lock_);
```

Comment:
IMHO having a data member that is not accessed by the current class (even not the c'tor) may be an indication of a code-smell.
What are the tradeoffs if the `LogicalHostDescription` class is completely moved into `LogicalHost` while the latter ensures that all accesses are done using the mutex?

(If there are later uses of the `LogicalHostDescription` ignore this comment)

Reply:
Your sense of smell was spot on. Really there's no need to have LogicalHost and LogicalHOstDescription be separate; they weren't before; I think in the process of doing this refactor it seemed necessary but it clearly isn't. 

Done.